mend_sca_cleanup_tool/sca_cleanup_tool.py

```diff
@@ -133,13 +124,8 @@ def delete_scan(product_token, project):
                 "projectToken": project['token'],
                 "agentInfo": AGENT_INFO
             })
-    r = post_api_request(request)
-    try:
-        response_obj = json.loads(r.decode("utf-8"))
-    except:
-        response_obj = json.loads(r)
-    if check_response_error(response_obj):
-        return
+    response_obj = try_or_error(lambda: json.loads(call_api(data=request)), {})
```

Comment:
With the try or error, What is the error that is returned when the API call fails? Is it the message in call_api? or is it the 2nd half of the lambda

Reply:
In this case {} will be returned (Second parameter in the try_or_error function)

Reply:
What is the benefit of the try_or_error if we are losing error feedback

Reply:
The tool did not fail; it just finished working.

Reply:
Correct. I changed the line 211. Pushed.

Reply:
So if we are keeping the try or error, why is the error portion empty? Versus just calling the function and letting it error? Feels like it is adding unneeded complexity

Reply:
We combine most try except constructs in one place. Not all, of course, but big part

Reply:
The problem with the empty result is the way the check_error_response function works. It needs the result to be a dict and that dict to contain a specific value to display the error message of the api call, otherwise it assumes the function passed. The point of the check_error_response is to check the api call after it succeeded. It was never supposed to reach that code if the api call failed.

`def check_response_error(obj_response):
    if isinstance(obj_response, dict):
        if "errorMessage" in obj_response:
            print(f"There was an issue with the request: {obj_response['errorMessage']}")
            return True
        else:
            return False`
            
      The Try - Error isn't reducing the amount of Try/Excepts in this script but it will eat the error from call_api which will never bubble up.
      
     I am not arguing that it is a good common solution but it isn't needed in this script and is hurting the functionality.

Reply:
Ok. I Returned try except construct according to existing logic in the tool.